### PR TITLE
Adding sqrt op in ttmlir e2e

### PR DESCRIPTION
**Pull request by sdjordjevicTT**
We need to implement the square root operation end-to-end for the next milestone of bringing up LLama.

Solves: https://github.com/tenstorrent/tt-mlir/issues/418

### CODE REVIEW

File: include/ttmlir/Dialect/TTNN/IR/TTNNOps.td [hunk 1]
```diff
@@ -63,6 +63,14 @@ class TTNN_ElementwiseOp<string mnemonic, list<Trait> traits = []> :
     let results = (outs Variadic<AnyRankedTensor>:$results);
 }
 
+class TTNN_ElementwiseUnaryOp<string mnemonic, list<Trait> traits = []> :
+    TTNN_ElementwiseOp<mnemonic, traits> {
```
Comment (svuckovicTT) [inline include/ttmlir/Dialect/TTNN/IR/TTNNOps.td:67]: The only thing I don't like here is that inputs and outputs are variadic for a unary op. Should we extend from `TTNN_NamedDPSOp` instead?

Comment (svuckovicTT) [reply include/ttmlir/Dialect/TTNN/IR/TTNNOps.td:67]: Didn't know that was an option, sounds great!

Comment (sdjordjevicTT) [reply include/ttmlir/Dialect/TTNN/IR/TTNNOps.td:67]: Thanks for the suggestion, I will make a default builder for unary op definitely!

Comment (sdjordjevicTT) [reply include/ttmlir/Dialect/TTNN/IR/TTNNOps.td:67]: Added custom builders for unary and binary element-wise ops!

**Review by nsmithtt — APPROVED** [review]
Awesome

**Comment by tapspatel** [comment]
You can add your sqrt.mlir to automatically generate ttnn files. I added docs on how to do that here: https://docs.tenstorrent.com/tt-mlir/ttrt.html#adding-llvm-lit-config-options-inside-a-mlir-file-to-create-flatbuffer-binaries. 

1. Copy your .mlir file into /test/ttmlir/Silicon/TTNN
2. Add relevant llvm-lit configs (for ex: `--ttir-load-system-desc="path=%system_desc_path%"`)

To test (see https://docs.tenstorrent.com/tt-mlir/ttrt.html#generate-flatbuffer-files-using-llvm-lit)
1. cmake --build build -- check-ttmlir
2. ttrt run build/test/ttmlir/Silicon/TTNN

Inside build/test/ttmlir/Silicon, you sqrt.ttnn file should've been generated.

**Review by tapspatel — APPROVED** [review]

**Review by svuckovicTT — APPROVED** [review]

**Comment by sdjordjevicTT** [comment]
> You can add your sqrt.mlir to automatically generate ttnn files. I added docs on how to do that here: https://docs.tenstorrent.com/tt-mlir/ttrt.html#adding-llvm-lit-config-options-inside-a-mlir-file-to-create-flatbuffer-binaries.
> 
> 1. Copy your .mlir file into /test/ttmlir/Silicon/TTNN
> 2. Add relevant llvm-lit configs (for ex: `--ttir-load-system-desc="path=%system_desc_path%"`)
> 
> To test (see https://docs.tenstorrent.com/tt-mlir/ttrt.html#generate-flatbuffer-files-using-llvm-lit)
> 
> 1. cmake --build build -- check-ttmlir
> 2. ttrt run build/test/ttmlir/Silicon/TTNN
> 
> Inside build/test/ttmlir/Silicon, your sqrt.ttnn file should've been generated.

Great, thanks for this @tapspatel

**Pull request merged**